src/Components/SharedComponents/FormTextField/FormTextField.tsx

```diff
@@ -0,0 +1,53 @@
+import { InputAdornment, TextField } from "@mui/material";
+import { ReactNode } from "react";
+import { FieldValues, RegisterOptions, UseFormRegister } from "react-hook-form";
+interface FormTextFieldType {
+  placeholder: string;
+  errors: any;
+  name: string;
+  register: UseFormRegister<FieldValues>;
+  rules?: RegisterOptions;
+  icon?: ReactNode;
+  type?: string;
+}
+export const FormTextField = ({
+  placeholder,
+  errors,
+  name,
+  register,
```

Comment:
I think you tried to use the register directly in the component and it didn't work. The problem is that the register function needs the ref to be able handle the input state an focus it when there is an error ...etc, but ref aren't forwarded to the components by default.
You can watch the lessons releated to refs and forwarding refs in this playlist, it talks about this issue.
https://youtube.com/playlist?list=PL75Ww3SxX-U9cx6yU_886qbahyp4Ilm1d&si=hxpsv94RKx0PZ6-6

Reply:
register is working without any error